Runtime/Utils/Values/SubscribableDictionaryValue.cs

```diff
@@ -0,0 +1,182 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Padoru.Core
+{
+    public enum DictionaryEvent
```

Comment:
This enum is the same as the one in `SubscribableListValue` which makes sense because they are both collections. Extract the enum to it's own script, rename it to CollectionEvents, and use it on both classes. Also rename the inner values to be more generic